Views/EditorTemplates/Parts/AdvancedSKUProduct.cshtml

```diff
@@ -4,36 +4,46 @@
 
 @{
     string skuHtml = "";
+    var divId = Html.IdForModel().ToString() + Guid.NewGuid().ToString();
 }
-@if (Model.Settings.AllowCustomPattern) {
-    //allow editing the sku
-    skuHtml = "<label class=\\\"sub\\\" for=\\\"" + Html.IdFor(m => m.CurrentSku) + "\\\">" + T("Sku") + "</label><br />" +
-        Html.TextBoxFor(m => m.CurrentSku, new { @class = "text" }).ToString().Replace("\"", "\\\"") + "<br />" +
-        "<span class=\\\"hint\\\">" + T("Save the current item and leave the input empty to have the sku automatically generated using the pattern {0}", Model.SkuPattern) + "</span>";
-}
-else {
-    if (Model.Settings.GenerateSKUAutomatically) { //don't allow editing the sku
-        if (string.IsNullOrWhiteSpace(Model.CurrentSku)) {
-            skuHtml = "<span class=\\\"\\\">" + T("Save the current item to have the sku automatically generated using the pattern {0}", Model.SkuPattern) + "</span>";
+<div id="@divId">
+    @if (Model.Settings.AllowCustomPattern) {
+        //allow editing the sku
+        @Html.LabelFor(m => m.CurrentSku, T("Sku").Text, new { @class = "sub" }) <br />
+        @Html.TextBoxFor(m => m.CurrentSku, new { @class = "text" })
+        <span class="hint">
+            @T("Save the current item and leave the input empty to have the sku automatically generated using the pattern {0}", Model.SkuPattern)
+        </span>
+    }
+    else {
+        if (Model.Settings.GenerateSKUAutomatically) { //don't allow editing the sku
+            if (string.IsNullOrWhiteSpace(Model.CurrentSku)) {
+                <span class="hint">
+                    @T("Save the current item and leave the input empty to have the sku automatically generated using the pattern {0}", Model.SkuPattern)
+                </span>
+            }
+            else {
+                <span><b>Sku:</b> @Model.CurrentSku</span>
+            }
         }
         else {
-            skuHtml = "<span><b>Sku:</b> " + Model.CurrentSku + "</span>";
+            //allow editing the sku (we are not generating the sku automatically)
+            @Html.LabelFor(m => m.CurrentSku, T("Sku").Text, new { @class = "sub" }) <br />
+            @Html.TextBoxFor(m => m.CurrentSku, new { @class = "text" })
         }
     }
-    else {
-        //allow editing the sku (we are not generating the sku automatically)
-        skuHtml = "<label class=\\\"sub\\\" for=\\\"" + Html.IdFor(m => m.CurrentSku) + "\\\">" + T("Sku") + "</label><br />" +
-            Html.TextBoxFor(m => m.CurrentSku, new { @class = "text" }).ToString().Replace("\"", "\\\"") + "<br />";
-    }
-}
+</div>
+
 
 @using (Script.Foot()) {
     <script type="text/javascript">
         //replace the sku portion of the usual view with what we have here
         $(function () {
+            var myDiv = $("#@divId");
+            myDiv.remove();
```

Comment:
I'm nitpicking at this point, but I would have instead used an inline style to hide the div, instead of relying on script to remove it.